server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -631,4 +630,24 @@ export class AssetService {
   getAssetCountByUserId(authUser: AuthUserDto): Promise<AssetCountByUserIdResponseDto> {
     return this._assetRepository.getAssetCountByUserId(authUser.id);
   }
+
+  async checkAssetsAccess(authUser: AuthUserDto, assetIds: string[], mustBeOwner = false) {
```

Comment:
The explicit `continues` in this function feel a bit awkward to me. I think it could be more readable to invert the checks and throw the ForbiddenException inside the `if`s, letting the `continue` happen implicitly if execution makes it past the checks. That might just be personal taste though.

Reply:
I understand what you mean. The problem is that if one of the checks fails, this does not mean that the access is forbidded, because one of the next checks may pass.

It's  `check1 OR check2 OR check3`, not `AND`.

Reply:
I think you're correct, but I'm currently having a little trouble evaluating the correctness of the checks (my brain is fried lol). Probably good to go, but @alextran1502 or @zoodyy would you mind taking another look at this function for me just to be sure?

Reply:
@bo0tzz I simplified the control flow and removed the continues. I think it's much clearer now

Reply:
Very nice! Thanks :)

Reply:
it's different now. but SLOWER:

- `getSharedWithUserAlbumCount` is more expensive then `countByIdAndUser`
- In the current version, the second check is needlessly performed even if the user owns the asset (first check successful is enough).

i actually liked the `continue` approach and just throwing a the end if no check succeeded.

Alternative: change ` if (!mustBeOwner) {` to `if (!accessAllowed && !mustBeOwner) {`. Still, I liked the continue better (less & clearer code to me)